src/ORM/Query.php

```diff
@@ -920,6 +925,26 @@ protected function _addDefaultFields()
     }
 
     /**
+     * Sets the default types for converting the fields in the select clause
+     *
+     * @return void
+     */
+    protected function _addDefaultSelectTypes()
+    {
+        $typeMap = $this->typeMap()->defaults();
+        $selectTypeMap = $this->selectTypeMap();
```

Comment:
Because they are not necessarily the same. More specifically, the select type map only contains the type for the aliases in the select clause. Sharing a single object for both would make the code quite difficult to manage.

A good example of not being the same is that you may want to add conditions using 'datetime' but at the end cast the column as a simple string.